### ISSUE TITLE
Support for Sitemap Response with text/html Content Type

### ISSUE THREAD

**Issue by smissingham** [TurnerSoftware/SitemapTools#85]
Thanks for the library James, it has proven very reliable and easy to use!

I have a tiny change request.

Many CMS plugins return sitemap files with Content-Type `text/html` (Magento, Wordpress, Drupal).
Though technically not correct, I cannot call my scraping targets and ask them to fix their sitemap...

This can be worked around by passing `HttpClient` into the `SitemapQuery` constructor and defining the `accept` header to `application/xml`, but that will only work for singular Sitemaps and not for the sites with many separate sitemaps.``

I cloned your repo locally, and added a single line to your `SitemapTypeMapping` dictionary in `SitemapQuery.cs`, and it all worked flawlessly:
```
			SitemapTypeMapping = new Dictionary<string, SitemapType>
			{
				{ "text/html", SitemapType.Xml }, // I added this line
				{ "text/xml", SitemapType.Xml },
				{ "application/xml", SitemapType.Xml },
				{ "text/plain", SitemapType.Text }
			};
```

Worth noting, this does assume that `text/html` returns XML type, but this is already wrapped in a HTTP 200 OK check, so I think it's a reasonably safe assumption.

Alternative to this option, maybe it would be good to allow developer to specify their own mapping, or to extend the list of parsers.

**Comment by smissingham** [comment]
If you're open to contributions, I'd be happy to look into submitting PR's for the proposed options, but you're currently not accepting PRs

**Comment by smissingham** [comment]
Changed `SitemapQuery.cs` to the following, and it was all happy:
```
if (response.IsSuccessStatusCode && SitemapTypeMapping.ContainsKey(response.Content.Headers.ContentType.MediaType))
```

**Comment by Turnerj** [comment]
Hey @smissingham - thanks for raising this issue! I'm not really sure there is a good solution for the problem there - if a website returns a `200` status code with a `text/html`, that could be like you've described or could be just an invalid media type but actually does contain a real sitemap file. It gets further complicated by the process of doing a HEAD or GET call and that `application/x-gzip` is also a valid response but not a valid sitemap type itself.

It is really more of a case though where you'd actually rather just not throw the exception for invalid mapping types? So somehow either toggle the behaviour (or just do a breaking change to remove the exception) and have `GetAllSitemapsForDomainAsync` ignore `null` values when it calls `GetSitemapAsync`.

> If you're open to contributions, I'd be happy to look into submitting PR's for the proposed options, but you're currently not accepting PRs

I am definitely open to pull requests! Happy to chat about the above more here first so you don't waste your own time though.

**Comment by smissingham** [comment]
Makes sense, and I like the alternative idea of switching the exceptions off during sitemap discovery process.

PR opened, keen to hear your thoughts. Thanks again.
